appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/sender/discovery/view/AppsDiscoveryActivity.kt

```diff
@@ -25,15 +28,21 @@ class AppsDiscoveryActivity : AppCompatActivity() {
         loader = findViewById(R.id.loader)
         findViewById<AppsDiscoveryList>(R.id.list).setLoadingListener(object : AppsDiscoveryListPresenter.LoadingListener {
             override fun loading() {
-                updateLoading()
+                handler.post {
```

Comment:
this shouldn't be done here. Make sure that in `DiscoverAppsRepository` where you execute the background thread, you always respond using the ui thread and then you won't need to wrap with a handler here.